File: Day-09-of-30-Days/Solution.cpp
// Day 09 of 30 Days 👾

#include <iostream>
using namespace std;

int sumOfDigits(int num) {
    // base case
    if (num < 10) {
        return num;
    }

    int sum = 0;
    while (num != 0) {
        sum += num % 10;
        num /= 10;
    }

    return sumOfDigits(sum);
}

// examples
int main() {
    cout << sumOfDigits(2345) << endl;
    return 0;
}

File: Day-09-of-30-Days/Solution.py
# Day 09 of 30 Days 👾

def sum_of_digits(num):
    # base case
    if num < 10:
        return num

    sum = 0;
    while num != 0:
        sum += num % 10
        num //= 10

    return sum_of_digits(sum)

# Example
print(sum_of_digits(2345))